std/experimental/ndslice/slice.d

```diff
@@ -1767,6 +1767,184 @@ struct Slice(size_t _N, _Range)
         assert(a != [[9, 2], [3, 4]]);
     }
 
+    /++
+    Computes hash value using MurmurHash3 algorithms without the finalization step.
+    Built-in associative arrays have the finalization step.
+
+    Returns: Hash value type of `size_t`.
+
+    See_also: $(LREF Slice.toMurmurHash3), $(MREF std, _digest, murmurhash).
+    +/
+    size_t toHash() const
+    {
+        static if (size_t.sizeof == 8)
+        {
+            auto ret = toMurmurHash3!128;
+            return ret[0] ^ ret[1];
+        }
+        else
+        {
+            return toMurmurHash3!32;
+        }
+    }
+
+    static if (doUnittest)
+    ///
+    pure nothrow @nogc @safe
+    unittest
+    {
+        import std.experimental.ndslice.selection : iotaSlice;
+        const sl = iotaSlice(3, 7);
+        size_t hash = sl.toHash;
+    }
+
+    static if (doUnittest)
+    ///
+    pure nothrow
+    unittest
+    {
+        import std.experimental.ndslice.iteration : allReversed;
+        import std.experimental.ndslice.selection : iotaSlice;
+
+        // hash is the same for allocated data and for generated data
+        auto a = iotaSlice(3, 7);
+        auto b = iotaSlice(3, 7).slice;
+
+        assert(a.toHash == b.toHash);
+        assert(typeid(typeof(a)).getHash(&a) == typeid(typeof(b)).getHash(&b));
+
+        // hash does not depend on strides
+        a = iotaSlice(3, 7).allReversed;
+        b = iotaSlice(3, 7).allReversed.slice;
+
+        assert(a.toHash == b.toHash);
+        assert(typeid(typeof(a)).getHash(&a) == typeid(typeof(b)).getHash(&b));
+    }
+
+    /++
+    Computes hash value using MurmurHash3 algorithms without the finalization step.
+
```

Comment:
The proper description of `size` and `opt` is large enough. Instead of Params I added reference to MMH3 in the `See_also` section.